### PR TITLE
En_St

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Completely different to the OoT version, so no documentation.

### CODE REVIEW

File: src/overlays/actors/ovl_En_St/z_en_st.h [hunk 1]
```diff
@@ -7,11 +7,50 @@ struct EnSt;
...
+#define ENST_GET_3F(thisx) (((thisx)->params & 0x3F) & 0xFF)
+#define ENST_GET_1C0(thisx) ((((thisx)->params & 0x1C0) >> 6) & 0xFF)
+
+enum {
+    /* 63 */ ENST_3F_63 = 63,
+};
+
+enum {
+    /* 1 */ ENST_1C0_1 = 1,
+};
```
[the remark this answers is not in the document]
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_St/z_en_st.h:19]: I know you said that before but Kenix gave me a review saying to add it even though it was only used with 1 value. I can change it either way but I need to know which.

[the remark this answers is not in the document]
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_St/z_en_st.h:19]: There doesn't seem to be any point making an enum for one thing: might as well use a define.

[the remark this answers is not in the document]
Comment (engineer124) [reply src/overlays/actors/ovl_En_St/z_en_st.h:19]: +1 on these being defines. There's no general rule of thumb we can provide, it all depends on the context of the actor. And in this case, enums doesn't really make sense

File: src/overlays/actors/ovl_En_St/z_en_st.c [hunk 2]
```diff
@@ -100,73 +155,789 @@ static DamageTable D_808A766C = {
...
+void func_808A52A8(EnSt* this, GlobalContext* globalCtx) {
+    static Color_RGBA8 D_808A7714 = { 170, 130, 90, 255 };
+    static Color_RGBA8 D_808A7718 = { 100, 60, 20, 0 };
+    s32 pad;
+    Vec3f spB8;
+    Vec3f spAC;
+    Vec3f spA0;
+    Vec3f sp94;
+    s32 i;
+    s16 temp_s0 = (Rand_ZeroOne() - 0.5f) * 65536.0f;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_St/z_en_st.c:210]:
```suggestion
    s16 temp_s0 = (Rand_ZeroOne() - 0.5f) * 0x10000;
```

File: src/overlays/actors/ovl_En_St/z_en_st.c [hunk 3]
```diff
@@ -100,73 +155,789 @@ static DamageTable D_808A766C = {
...
+            func_808A576C(this);
+            this->unk_318 = 10;
+        }
+    }
+}
+
+void EnSt_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnSt* this = THIS;
+
+    this->unk_2C0 = Object_GetIndex(&globalCtx->objectCtx, 1);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_St/z_en_st.c:838]:
```suggestion
    this->unk_2C0 = Object_GetIndex(&globalCtx->objectCtx, GAMEPLAY_KEEP);
```

File: src/overlays/actors/ovl_En_St/z_en_st.h [hunk 4]
```diff
@@ -7,11 +7,50 @@ struct EnSt;
...
+    /* 0x310 */ s16 unk_310;
+    /* 0x312 */ s16 unk_312;
+    /* 0x314 */ s16 unk_314;
+    /* 0x316 */ s16 unk_316;
+    /* 0x318 */ s16 unk_318;
+    /* 0x31A */ s16 unk_31A;
+    /* 0x31C */ s16 unk_31C[12];
+    /* 0x334 */ s16 unk_334[12];
+    /* 0x34C */ f32 unk_34C;
+    /* 0x350 */ char unk_350[0x8];
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_St/z_en_st.h:50]:
```suggestion
    /* 0x350 */ UNK_TYPE1 unk_350[0x8];
```

**Review by engineer124 — COMMENTED** [review]
Good work. Just a small number of comments

**Review by AngheloAlf — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**